docs: add v2020.2 release notes (#2062)
freifunk-gluon/gluon: docs/releases/v2020.2.rst

Gluon 2020.2
=============

Added hardware support
----------------------

ath79-generic
~~~~~~~~~~~~~

* GL.iNet

  - GL-AR750S

* TP-Link

  - CPE220 (v3)

ipq40xx-generic
~~~~~~~~~~~~~~~

* EnGenius

  - ENS620EXT [#outdoor]_

* Linksys

  - EA6350 (v3)

ramips-mt76x8
~~~~~~~~~~~~~
* Cudy

  - WR1000

lantiq-xway
~~~~~~~~~~~

* NETGEAR

  - DGN3500B

lantiq-xrx200
~~~~~~~~~~~~~

* TP-Link

>   - TD-W8970

Reviewer: Alphabetical order:
- lantiq-xrx200
- lantiq-xway
- ramips-mt76x8

x86-legacy [#newtarget]_
~~~~~~~~~~

* Devices older than the Pentium 4

.. [#newtarget]
  This is a new target in this release

> .. [#oudoor]

Reviewer 2: ```suggestion
.. [#outdoor]
```

>   This device is supposed to be setup outdoors and will therefore have it's outdoor mode flag automatically enabled.

Reviewer 2: ```suggestion
  This device is supposed to be set up outdoors and will therefore have it's outdoor mode flag automatically enabled.
```

Reviewer: "it's" -> "its"

Major changes
-------------

Device Classes
==============

> Devices not can be categorised into device classes. This device class can determine which features

Reviewer 3: ```suggestion
Devices now can be categorised into device classes. This device class can determine which features
```
typo

Reviewer: Hmm, that's not quite accurate either, as it's already done. "Devices have been categorized into device classes."?

Reviewer 4: Why not keep it simple: "Devices are now categorized into device classes."?

as well as packages are installed on the device when building images.

> Yurrently there are two classes used in Gluon, tiny and standard. All devices with less than 64M of Memory or

Reviewer 3: ```suggestion
Currently there are two classes used in Gluon, tiny and standard. All devices with less than 64M of Memory or
```
typo

Reviewer: - Another typo: "Memory" -> "memory" (or maybe better write "RAM" to make this clearer)
- "tiny" and "standard": quote or enclose in `*`

> 7M of usable firmware space are assigned to the tiny class.

Reviewer: Style: repeat the "less than": "less than 64MiB of RAM or less than 7MiB of usable firmware space"

> WPA3-Support for Private WLAN

Reviewer: "WPA3-Support" -> "WPA3 support"

=============================

The private WLAN now supports WPA3-SAE key exchange as well as management frame protection (802.11w).

> For this to work, the firmware needs to be built with the wireless-encryption-wpa3 feature.

Reviewer: Enclose wireless-encryption-wpa3 in `*`

Enhanced Open on Client Network
===============================

* 59a4cd63 gluon-respondd: expose OWE clients in nodeinfo
* 6692095f gluon-client-bridge: add support for OWE encryption

SAE Encrypted Mesh Links
========================
* 461d9040 Add gluon-mesh-wireless-sae package

> Responsive Status-Page

Reviewer: "Responsive status page"

======================

The status page design is now responsive and reflows better on mobile devices.

> Primary Domain-Code

Reviewer: "Primary domain code"

===================

The primary domain code is now visible on the node status page as well as in the respondd information
emitted by the node.

Logging
=======

- 9b19883d add gluon-logging package
- b181803a scripts: check_site: add need_number_range check

Peer cleanup in fastd
=====================

- 3ccf7fdd gluon-mesh-vpn-fastd: clean up peers and groups on update

Bugfixes
--------

> - The WAN MAC address now matches the one defined in OpenWrt if VXLAN is enabled for the selected domain.

Reviewer: Hmm, I'm sure we have a few more bugfixes... I'll have another look at the history

Reviewer: - 3d091912af41 gluon-web-wifi-config: make mesh VIF options depend on outdoor=false
- 22847e49867f gluon-reload: add missing service restarts on domain switching

Site changes
------------

site.mk
=======

> * OpenWrt since 19.07 ships the urngd entropy daemon by default. It replaces the haveged daemon, for which we removed the support in Gluon. Remove ``haveged`` from your package selection.

Reviewer 2: ```suggestion
* Starting with version 19.07 OpenWrt ships the urngd entropy daemon by default. It replaces the haveged daemon, for which we removed the support in Gluon. Remove ``haveged`` from your package selection.
```

Internal
--------

Editorconfig
* 1c2dc51b add .editorconfig (#1951)

> Continous Integration

Reviewer 3: ```suggestion
Continuous Integration
```
typo

Gluon 2020.2
=============

Added hardware support
----------------------

ath79-generic
~~~~~~~~~~~~~

* GL.iNet

  - GL-AR750S

* TP-Link

  - CPE220 (v3)

ipq40xx-generic
~~~~~~~~~~~~~~~

* EnGenius

  - ENS620EXT [#outdoor]_

* Linksys

  - EA6350 (v3)

lantiq-xrx200
~~~~~~~~~~~~~

* TP-Link

  - TD-W8970

lantiq-xway
~~~~~~~~~~~

* NETGEAR

  - DGN3500B

ramips-mt76x8
~~~~~~~~~~~~~
* Cudy

  - WR1000

x86-legacy [#newtarget]_
~~~~~~~~~~~~~~~~~~~~~~~~

* Devices older than the Pentium 4

.. [#newtarget]
  This is a new target.

.. [#outdoor]
  This device is supposed to be set up outdoors and will therefore have its outdoor mode flag automatically enabled.

Major changes
-------------

Device Classes

> ==============

Reviewer: ```suggestion
~~~~~~~~~~~~~~
```
`====` is already being used for the page title. This issue appears many more times - which can also be seen in the table of contents / index page.

Devices are now categorized into device classes. This device class can determine which features
as well as packages are installed on the device when building images.

Currently there are two classes used in Gluon, *tiny* and *standard*. All devices with less than 64M of RAM or
less than 7M of usable firmware space are assigned to the tiny class.

WPA3 support for Private WLAN
=============================

The private WLAN now supports WPA3-SAE key exchange as well as management frame protection (802.11w).
For this to work, the firmware needs to be built with the *wireless-encryption-wpa3* feature.

OWE on Client Network
=====================

Gluon now allows to configure a VAP for the client network which supports opportunistic encryption on the
client network for devices which support the OWE security type (also known as Enhanced Open).

This encrypted VAP can be the only available access point or be configured in addition to an unencrypted VAP.
In the latter case, the transition mode can be enabled, which enables compatible devices to automatically

> connect to the encrypted VAP while legacy devices continue to use the unencrypted connection.

Reviewer 5: 1) i think "in the latter case" is wrong. isn't the transition mode only helpful if there is only one VAP? the "latter case" is the case with two VAPs, though?

2) @NeoRaider the issue is not really about "old" Android devices, only about some specific Android 9 devices which have a bug related to transition mode which leads to them not being able to connect if you configure OWE like you would want it in an ideal world: only one SSID with active transition mode - for Freifunk this means that you need a separate SSID like "Freifunk OWE" for years to come to offer this (or you do it the other way around, default SSID has transition mode, separate SSID without OWE for broken devices)

> connect to the encrypted VAP while legacy devices continue to use the unencrypted connection.

Reviewer 5: looks like this is the only open discussion before we can finish the release? @mweinelt @blocktrron

> connect to the encrypted VAP while legacy devices continue to use the unencrypted connection.

Author: @rotanid sorry, i've had this comment pending and forgot to submit it

Bugfixes
--------

- The WAN MAC address now matches the one defined in OpenWrt if VXLAN is enabled for the selected domain.

- *gluon-reload* now reloads all relevant services.

- Disable outdoor mode and enabling meshing in the config mode can now be performed in a single step.

- Fixed section visiblity with enabled outdoor mode in config mode.

Site changes
------------

site.mk
=======

* Starting with version 19.07 OpenWrt ships the urngd entropy daemon by default.
  It replaces the haveged daemon, for which we removed the support in Gluon. Remove ``haveged`` from your package selection.

Internal
--------

Editorconfig
============

Gluon now ships a *editorconfig* file to allow compatible editors to automatically apply key aspects of Gluons code style.

Continuous Integration
======================

* Jenkins

  - The CI now has a test stage to verify Gluons runtime functionality.

* GitHub Actions

  - GitHub actions is now enabled for the Gluon project, build-testing all available targets.

Build system
============

  - Source code minification can now be skipped by enabling the GLUON_MINIFY flag.

>   - Enabling the GLUON_AUTOREMOVE flag will remove package build ditrectories after they are built.

Reviewer 6: There is a typo in "ditrectories".

    This reduces space consumption at the expense of subsequent builds being slower.

Known issues
------------

* Out of memory situations with high client count on ath9k.
  (`#1768 <https://github.com/freifunk-gluon/gluon/issues/1768>`_)

* The integration of the BATMAN_V routing algorithm is incomplete.

>    - | Mesh neighbors don't appear on the status page. (`#1726 <https://github.com/freifunk-gluon/gluon/issues/1726>`_)
>      | Many tools have the BATMAN_IV metric hardcoded, these need to be updated to account for the new throughput
>      | metric.
> 
>    - | Throughput values are not correctly acquired for different interface types.
>      | (`#1728 <https://github.com/freifunk-gluon/gluon/issues/1728>`_)
>      | This affects virtual interface types like bridges and VXLAN.

Reviewer: ```suggestion
  - Mesh neighbors don't appear on the status page. (`#1726 <https://github.com/freifunk-gluon/gluon/issues/1726>`_)
    Many tools have the BATMAN_IV metric hardcoded, these need to be updated to account for the new throughput
    metric.

  - Throughput values are not correctly acquired for different interface types.
    (`#1728 <https://github.com/freifunk-gluon/gluon/issues/1728>`_)
    This affects virtual interface types like bridges and VXLAN.
```
Putting the sub-points into a blockquote may be a nice hack to fix the broken margins in the RTD theme, but I think it would be better if we could just fix the CSS (which I hope is possible on readthedocs.io, and is also unrelated the release notes)

Gluon 2020.2
============

Added hardware support
----------------------

ath79-generic
~~~~~~~~~~~~~

* GL.iNet

  - GL-AR750S

* TP-Link

  - CPE220 (v3)

ipq40xx-generic
~~~~~~~~~~~~~~~

* EnGenius

  - ENS620EXT [#outdoor]_

* Linksys

  - EA6350 (v3)

lantiq-xrx200
~~~~~~~~~~~~~

* TP-Link

  - TD-W8970

lantiq-xway
~~~~~~~~~~~

* NETGEAR

  - DGN3500B

ramips-mt76x8
~~~~~~~~~~~~~
* Cudy

  - WR1000

x86-legacy [#newtarget]_
~~~~~~~~~~~~~~~~~~~~~~~~

> * Devices older than the Pentium 4

Reviewer 7: OpenWrt compiles with "-march=pentium-mmx" by default. https://github.com/openwrt/openwrt/pull/3061

My idea:
```
* x86-legacy is compatible with CPUs implementing at least pentium-mmx compatibility level. Everything older than that is not supported. This was implemented cause x86-generic compiles to pentium4 compatibility.
```
